projects/laji/src/app/shared-modules/laji-map/laji-map.component.ts

```diff
@@ -166,6 +119,32 @@ export class LajiMapComponent implements OnDestroy, OnChanges {
     if (changes.data) {
       this.setData(this.data);
     }
+
+    if (changes.options || changes.settingsKey) {
+      if (this.updateSettingsSub) {
+        this.updateSettingsSub.unsubscribe();
+      }
+
+      if (changes.options) {
+        this.updateOptions();
+      }
+
+      let settings$ = of(this.userSettings);
+
+      if (changes.settingsKey) {
+        const key = this.settingsKey;
```

Comment:
This could still be little refactored. Looks like `this.settingsKey` and `this._settingsKey` has always the same value so `_settingsKey` can be removed and also the `key` variable can be removed as it's just the `settingsKey`